File: pymuscle/SymbolicTensor.c
void SymbolicTensor(double I[4],
                    const double sx,
                    const double sy,
                    const double sz,
                    const double rho) {
    I[0] = rho*(sx*sx*sx)*sy*sz*(1.0/1.2E1);
	I[1] = rho*sx*(sy*sy*sy)*sz*(1.0/1.2E1);
	I[2] = rho*sx*sy*(sz*sz*sz)*(1.0/1.2E1);
	I[3] = rho*sx*sy*sz;
}

void BoxCorners(double corners[8][3], const double sx, const double sy, const double sz) {
    int i;
    for (i=0; i<8; ++i) {
        corners[i][0] = (1 - 2*( (i&4)>>2  )) * sx/2;
        corners[i][1] = (1 - 2*( (i&2)>>1  )) * sy/2;
        corners[i][2] = (1 - 2*(  i&1      )) * sz/2;
    }
}

File: Src/pymuscle/scripts/SymbolicTensor.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
"""
Optimization-based rigid body simulator
2010 Geoyeob Kim

Inertia tensor matrix defined in [Jain 2009].
For box shaped body.
"""
from numpy import *
from math import pow, sin, cos

def SymbolicTensor(sx, sy, sz, rho):
	Ixx = rho*(sx*sx*sx)*sy*sz*(1.0/1.2E1);
	Iyy = rho*sx*(sy*sy*sy)*sz*(1.0/1.2E1);
	Izz = rho*sx*sy*(sz*sz*sz)*(1.0/1.2E1);
	Iww = rho*sx*sy*sz;
	return Ixx, Iyy, Izz, Iww
	
